constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/LinearProfile.java

```diff
@@ -1,203 +1,134 @@
 package gov.nasa.jpl.aerie.constraints.model;
 
 import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.IntervalMap;
 import gov.nasa.jpl.aerie.constraints.time.Segment;
 import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
 import org.apache.commons.lang3.tuple.Pair;
 
 import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 
-public final class LinearProfile implements Profile<LinearProfile> {
+public final class LinearProfile implements Profile<LinearProfile>, Iterable<Segment<LinearEquation>> {
   // IMPORTANT: Profile pieces must be non-overlapping, and increasing (based on interval field)
-  public final List<LinearProfilePiece> profilePieces;
+  public final IntervalMap<LinearEquation> profilePieces;
 
-  public LinearProfile(final List<LinearProfilePiece> profilePieces) {
+  public LinearProfile(final IntervalMap<LinearEquation> profilePieces) {
     this.profilePieces = Objects.requireNonNull(profilePieces);
   }
 
-  public LinearProfile(final LinearProfilePiece... profilePieces) {
-    this(List.of(profilePieces));
+  @SafeVarargs
+  public LinearProfile(final Segment<LinearEquation>... profilePieces) {
+    this.profilePieces = IntervalMap.of(profilePieces);
```

Comment:
Thoughts on having these two constructors delegate to `this(profilePieces)`?

Reply:
No opinion. I've made the change